src/PointwiseFunctions/AnalyticData/NewtonianEuler/ShuOsherTube.hpp

```diff
@@ -0,0 +1,184 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticData/AnalyticData.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace NewtonianEuler::AnalyticData {
+/*!
+ * \brief Initial data for the Shu-Osher oscillatory shock tube \cite Shu1988439
+ *
+ * The general initial data is given by:
+ *
+ * \f{align*}{
+ *  \{\rho,v^x,p\}=
+ * \left\{
+ * \begin{array}{ll}
+ *   \left\{\rho_L, v^x_L, p_L\right\} &\mathrm{if} \;\;\; x<\Delta \\
+ *   \left\{1+\epsilon\sin(\lambda x), v^x_R, p_R\right\} &\mathrm{if} \;\;\;
+ *    x\ge \Delta
+ * \end{array}\right.
+ * \f}
+ *
+ * with the adiabatic index being 1.4.
+ *
+ * With the standard parameters given below, this is a Mach-3 shock moving into
+ * a sinusoidal density profile.
+ *
+ * \f{align*}{
+ *  \{\rho,v^x,p\}=
+ * \left\{
+ * \begin{array}{ll}
+ *   \left\{3.857143, 2.629369, 10.33333\right\} &\mathrm{if} \;\;\; x<-4 \\
+ *   \left\{1+0.2\sin(5x), 0, 1\right\} &\mathrm{if} \;\;\; x\ge -4
+ * \end{array}\right.
+ * \f}
+ *
+ * With these values the usual final time is 1.8.
+ */
+class ShuOsherTube : public MarkAsAnalyticData {
+ public:
+  using equation_of_state_type = EquationsOfState::IdealFluid<false>;
+
```

Comment:
Does this class need a `source_term_type` alias to run with the NewtonianEuler executable?

Reply:
Not sure, probably... I'll add it just in case. We can't test that until we have the new boundary conditions, I think, because currently we can only impose analytic solutions as boundary conditions...